tutorials/lab/coral2.rst

```diff
@@ -18,96 +18,82 @@ Things to Know
-
-  wget https://github.com/flux-framework/flux-core/releases/download/v0.29.0/flux-core-0.29.0.tar.gz
-
-  tar -xzvf flux-core-0.29.0.tar.gz && cd flux-core-0.29.0
-
-  ./configure --prefix=$PREFIX && make -j && make install && cd ..
-
-  # The `flux` executable will now be in ~/local/bin/flux but it needs some
-  # additional flux-coral2 extensions
-
-
-  # Step 2: Build flux-sched 0.18 or later (optional but recommended)
 
-  wget https://github.com/flux-framework/flux-sched/releases/download/v0.18.0/flux-sched-0.18.0.tar.gz
+------------------
+Flux with Cray PMI
+------------------
 
-  tar -xzvf flux-sched-0.18.0.tar.gz && cd flux-sched-0.18.0
+Applications linked to Cray MPICH will work natively with Flux
+provided the Cray MPICH library uses the PMI2 protocol instead of
+the homespun Cray PMI and libPALS. For Flux to support libPALS,
+flux-coral2 must be built (see above) and Flux must be configured
+to offer libPALS support. This is done by setting the "pmi" shell
+option to include "cray-pals" on a per-job basis like so:
 
-  ./configure --prefix=$PREFIX && make -j && make install && cd ..
+``flux submit -n1 -opmi=cray-pals hostname``
 
+or by configuring Flux to offer such support by default, by adding
+the following lines to the shell's ``initrc.lua`` file:
 
-  # Step 3: Build flux-coral2
+.. code-block:: lua
 
-  git clone https://github.com/flux-framework/flux-coral2.git && cd flux-coral2
+    if shell.options['pmi'] == nil then
+        shell.options['pmi'] = 'cray-pals,simple'
+    end
 
-  ./autogen.sh && ./configure --prefix=$PREFIX && make -j && make install
 
-  libtool --finish $PREFIX/lib/flux/job-manager/
-  libtool --finish $PREFIX/lib/flux/shell/plugins/
-  cd ..
+The lines should come before any call to load plugins.
 
+If Flux jobs that use Cray MPICH end up as a collection of singletons,
+that is usually a sign that Cray MPICH is trying to use libPALS.
 
-  # Step 4: add a config file to automatically load a flux-coral2 plugin
+-----------------------------
+Configuring Flux with Rabbits
+-----------------------------
 
-  mkdir -p $PREFIX/etc/flux/config
+In order for a Flux system instance to be able to allocate
+rabbit storage, the ``dws_jobtap.so`` plugin must be loaded.
+The plugin can be loaded in a  config file like so:
 
-  echo "[job-manager]
-  plugins = [
-     { load = \"cray_pals_port_distributor.so\", conf = { port-min = $PORT_MIN, port-max = $PORT_MAX } }
-  ]
-  " > $PREFIX/etc/flux/config/cray_pals_ports.toml
+.. code-block::
 
-  echo "Done! Now set FLUX_CONF_DIR=$PREFIX/etc/flux/config
-  in your environment and run with $PREFIX/bin/flux"
+    [job-manager]
+    plugins = [
+      { load = "dws-jobtap.so" }
+    ]
 
+Also, the ``flux-coral2-dws`` systemd service must be started
+on the same node as the rank 0 broker of the system instance
+(i.e. the management node). The ``flux`` user must have
+a kubeconfig file in its home directory granting it read
+and write access.
```

Comment:
Should this end with something like "You can read more about Flux with rabbit storage here (link to the rabbit doc below?)

Reply:
I think it's OK without that transition. This document is for administrators, the other is for users.